tests/server/test_server.py

```diff
@@ -57,31 +57,36 @@ def repeat_value(self, payload):
         self._host.reply({"value": payload["value"] * 3})
 
 
-def TestServerBase(class_name, class_parents, class_attr):
+class TestServerBase(unittest.TestCase):
```

Comment:
There's a lot of noise here, but that 's simply refactoring so TestServerBase can be reused for the new unit tests. I recommend scrolling down to the bottom of the file to see the new tests.